src/main/kotlin/org/ptss/support/infrastructure/repositories/CategoryRepository.kt

```diff
@@ -0,0 +1,110 @@
+package org.ptss.support.infrastructure.repositories
+
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.transaction.Transactional
+import org.ptss.support.domain.interfaces.repositories.ICategoryRepository
+import org.ptss.support.domain.models.Category
+import org.ptss.support.infrastructure.persistence.entities.CategoryEntity
+import org.ptss.support.infrastructure.persistence.entities.ToolEntity
+import java.util.*
+
+@ApplicationScoped
+class CategoryRepository @Inject constructor(
+    private val entityManager: EntityManager
+) : ICategoryRepository, PanacheRepository<CategoryEntity> {
+    @Transactional
+    override suspend fun getAll(): List<Category> {
+        return entityManager.createQuery(
+            """
+            SELECT DISTINCT c 
+            FROM CategoryEntity c 
+            LEFT JOIN FETCH c.tools
+            """, CategoryEntity::class.java
+        )
+            .resultList
+            .map { it.toDomain() }
+    }
```

Comment:
same as comment below:
```suggestion
    override suspend fun getAll(): List<Category> {
        return find("FROM CategoryEntity c LEFT JOIN FETCH c.tools")
            .list()
            .map { it.toDomain() }
    }
```
if you use these suggestions; change your Icategory repository code and test it pls